auth/extensions.py

```diff
@@ -0,0 +1,26 @@
+from importlib import import_module
+from importlib.util import find_spec
+
+import logging
+
+from .credentials import allowed_extensions
+
+
+def action_user(extension, user_info={}):
+    if extension not in allowed_extensions:
+        return False
+
+    module_name = allowed_extensions[extension]
```

Comment:
split this logic to another function.

As a test, add another method here - 'on_login' which is called whenever a user logs in (regardless of new or old)